### PR TITLE
script for processing coconut smiles into a LIFT style train set

### PR DESCRIPTION
**Pull request by bethanyconnolly**
I have tested this quite a bit on a smaller subset of the data and I'm pretty sure it works correctly, but it would be good to get a second opinion in particular on the `concatenate_samples_without_splitting` function.
The main problem is that iteratively using rdkit makes `process_valid_smiles` and `process_invalid_smiles` extremely slow, I used just a couple of samples for testing 😅

### CODE REVIEW

**Comment by bethanyconnolly** [comment]
Made some improvements and it runs much faster now

File: experiments/data/prepare_smiles_dataset.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+DATA_TYPE = "text"
+EOS_TOKEN = "\n\n"
+SEED = 1234
+
+
+def process_docs(docs):
+    valid = map(_get_smile_string, docs)
+    invalid = map(_process_invalid_smile, docs)
+    mixed_data = list(valid) + list(invalid)
+    random.seed(seed=SEED)
```
Comment (jackapbutler) [inline experiments/data/prepare_smiles_dataset.py:36]: I think this is a duplicate of L108 in `run()`?

File: experiments/data/prepare_smiles_dataset.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        pretrained_model_name_or_path=config.model_name,
+        model_max_length=config.context_length,
+    )
+    if not tokenizer.pad_token:
+        tokenizer.add_special_tokens({"pad_token": "<|padding|>"})
+
+    batched_data = concatenate_samples_without_splitting(
+        mixed_data, tokenizer, config.context_length
+    )
+    total_tokens = sum([len(batch) for batch in batched_data])
```
Comment (jackapbutler) [inline experiments/data/prepare_smiles_dataset.py:128]: Does this get # samples instead of # tokens? as it's summing the length of each batch

**Review by jackapbutler — CHANGES_REQUESTED** [review]

File: experiments/data/prepare_smiles_dataset.py (hunk 2 above)
Comment (bethanyconnolly) [reply experiments/data/prepare_smiles_dataset.py:128]: Each batch in batched_data is a list which contained many samples concatenated together up to a maximum length of 2048 tokens. This total tokens is the total tokens in all the batches before any padding is added.

**Review by jackapbutler — APPROVED** [review]

**Pull request merged**